main.nf

```diff
@@ -18,10 +18,26 @@ nextflow.enable.dsl = 2
 
 WorkflowMain.initialise(workflow, params, log)
 
-include {METAGENOMICS_SR} from './workflows/metagenomics-sr'
+if (params.platform == 'illumina') {
+    include { ILLUMINA } from './workflows/illumina'
+} else if (params.platform == 'nanopore') {
+    include { NANOPORE } from './workflows/nanopore'
+}
+
+
+workflow ARCADIASCIENCE_METAGENOMICS {
+    //
+    // WORKFLOW: Illumina processing and assembly
+    //
+    if (params.platform == 'illumina') {
+        ILLUMINA ()
```

Comment:
this is a placeholder for the eventual workflow that will be executed when the param illumina is specified? (same comment for below `NANOPORE ()`

Reply:
or is this the pointer to the subworkflow...?

Reply:
the workflow names are already renamed to illumina and nanopore, so the illumina workflow was metagenomics-sr but I changed it to make it more explicit. The nanopore one is a placeholder because nothing is in that workflow yet so the flag for that platform would error out. This is sort of the same concept as subworkflows but not entirely because I use different subworkflows in these different platform workfows (such as the mapping_depth subworkflow used in the illumina workflow for bowtie2 mapping and depth calculation with metabat2 jgisummarizecontigs)

Reply:
Great questions from Taylor. I actually hate how `nf-core` tools initializes the pipeline structure. Either I am stupid and can't understand the difference between workflows or subworkflows, or the way `nf-core` does this is insanely convoluted, which I plan to change for the the phylorthology pipeline. 

As an example in the `phylorthology`, we have a single `main.nf` which then calls a singular file under `workflows` and all of their names are soooo long and convoluted that pipeline processes look like the attached image. 
<img width="545" alt="Screenshot 2023-01-21 at 5 13 12 PM" src="https://user-images.githubusercontent.com/2692053/213889003-297c0085-8026-4863-a2b2-b04d080c771c.png">

We can clean this up later here, but my gut feel is we should have `main.nf` as is and `ILLUMINA` and `NANOPORE` should be subworkflows and we get rid of the `workflows` folders.